charts/retool/templates/deployment_workflows.yaml

```diff
@@ -66,12 +66,28 @@ spec:
             value: {{ template "retool.deploymentTemplateVersion" . }}
           - name: NODE_ENV
             value: production
-          {{ if ( not $.Values.dbconnector.java.enabled ) }}
+          {{- $retool_version_with_java_dbconnector_opt_out := ( or (eq "latest" $.Values.image.tag ) ( semverCompare ">= 3.93.0-0" $.Values.image.tag ) ) }}
+          {{- $serviceType := list "WORKFLOW_BACKEND" "DB_CONNECTOR" "DB_SSH_CONNECTOR" }}
+          {{- /*
+            JAVA_DBCONNECTOR in the service type only applies before the version of Retool that changes it to opt-out (3.93.0-edge),
+            and only if the Java dbconnector is enabled in values.yaml.
+          */}}
+          {{- if and ( not $retool_version_with_java_dbconnector_opt_out ) ( $.Values.dbconnector.java.enabled ) }}
+            {{- $serviceType = append $serviceType "JAVA_DBCONNECTOR" }}
+          {{- end }}
+          {{- /*
+            It may seem counterintuitive to add the JOBS_RUNNER service type only without a jobs runner.
+            The reason for this is that the backend needs to act as a jobs runner, if the jobs runner is not enabled.
+          */}}
+          {{- if not .Values.jobRunner.enabled }}
+            {{- $serviceType = append $serviceType "JOBS_RUNNER" }}
```

Comment:
wait, the workflow backend should never have the jobs runner service. it wasn't in the original list either

Reply:
whoops